apps/test/unit/lib/kits/maker/MicroBitBoardTest.js

```diff
@@ -0,0 +1,71 @@
+import {expect} from '../../../../util/deprecatedChai';
+import MicroBitBoard from '@cdo/apps/lib/kits/maker/MicroBitBoard';
+import {MicrobitStubBoard} from './makeStubBoard';
+import sinon from 'sinon';
+
+describe('MicroBitBoard', () => {
+  let board;
+
+  beforeEach(() => {
+    // Construct a board to test on
+    window.SerialPort = {};
+    board = new MicroBitBoard();
+    board.boardClient_ = new MicrobitStubBoard();
+  });
+
+  afterEach(() => {
+    board = undefined;
+  });
+
+  describe(`connect()`, () => {
+    it('initializes a set of components', () => {
+      return board.connect().then(() => {
+        expect(Object.keys(board.prewiredComponents_)).to.have.length(6);
+        expect(board.prewiredComponents_.board).to.be.a('object');
```

Comment:
Is there anything else we can/should assert about the components other than that they're objects?

Reply:
The functionality of each component is covered in it's own test and the MicroBitComponentsTest covers creating, enabling, and clearing the components. The main purpose of the components in the MicroBitBoard file is that we receive them upon connect and then pass them through to different functions. Given that we don't interact with them other than as a 'do they exist/not' and passing as parameters and that their individual functionality is covered in different tests, I determined just checking for their presence to be sufficient here. Happy to discuss further if you have concerns!

Reply:
Makes sense, thanks!